client/webserver/site/src/js/forms.js

```diff
@@ -298,31 +302,91 @@ export class WalletConfigForm {
  * ConfirmRegistrationForm should be used with the "confirmRegistrationForm" template.
  */
 export class ConfirmRegistrationForm {
-  constructor (application, form, { getDexAddr, getCertFile }, success, insufficientFundsFail) {
+  constructor (application, form, { getDexAddr, getCertFile }, success, insufficientFundsFail, setupWalletFn) {
     this.fields = Doc.parsePage(form, [
-      'feeDisplay', 'marketRowTemplate', 'marketsTableRows', 'appPass', 'appPassBox',
-      'appPassSpan', 'submitConfirm', 'regErr'
+      'marketRowTemplate', 'marketsTableRows', 'appPass', 'appPassBox',
+      'appPassSpan', 'submitConfirm', 'regErr', 'feeRowTemplate',
+      'feeTableRows', 'setupWallet'
     ])
     app = application
     this.getDexAddr = getDexAddr
     this.getCertFile = getCertFile
     this.success = success
     this.insufficientFundsFail = insufficientFundsFail
     this.form = form
-    bind(form, this.fields.submitConfirm, () => this.submitForm())
+    this.setupWalletFn = setupWalletFn
+    this.feeWallet = null
+    bind(form, this.fields.submitConfirm, () => this.submitForm(this.feeWallet))
+  }
+
+  setWalletFee (assetID) {
+    this.feeWallet = assetID
   }
```

Comment:
Lots going on with the naming here. Is it a fee or a wallet or an asset ID? How about renaming `feeWallet` to `feeAssetID` or similar, and getting rid of `setWalletFee` and just set the field directly.